### ISSUE TITLE
DK - Gargoyle ... ?

### ISSUE THREAD

**Issue by Eradicatepls** [gamefreedomgit/Frostmourne#1519]
I think there is another issue, can't really try it on a raid environement, and don't really know where it's coming from, but gargoyle is basically nerfed to the ground, i've diverse exemple of a gargoyle perfectly snapshotted with all haste buffs possible in the game : Racial troll + glvs + pot + Lust. Barely 100K to 120k. With beetween 39 and 41 hit. 
So what's the matter ? Is gargoyle not converting strength anymore ? Because after gargoyle 0.2 ms fix, we had gargoyle with 180k ish damage, pushing to 200k with perfect rng with troll racial. 
So is it because of https://github.com/gamefreedomgit/Frostmourne/issues/1467 hotfix, or coming from somewhere else, if we can have a little help from the big brainer here.

**Comment by Eradicatepls** [comment]
Good snap with 20% haste and 10% ap in Raid + Good RNG : https://gyazo.com/54a3fd2ce67fdef33249b19b83acb419 // https://gyazo.com/a9b9a708c8ba36e223fa041bdf3cdd11

A Godly Pixel snap : https://gyazo.com/ec225e5982e4b170671bab471901fcee // https://gyazo.com/2ca55c820bcdf6f2b83d1617593837dd

I'll try to give more exemples later but it's pretty obvious that gargoyle is really in a bad shape now.

**Comment by Manpower99** [comment]
Potentially a reduction on the current 200 MS delay to 100 MS perhaps? @RadeghostWM 
A good point was raised here, Army got fixed to blizzlike state and Pet Ghoul got fixed to blizzlike state, a slight improvement to gargoyle would be a nice change prior to ulduar release.

**Comment by Eradicatepls** [comment]
I agree with you but Bro at this point we can just let gargoyle blizzlike, i mean why is the point in nerfing it, yeah dk is TOP 2 wotlk class in ulduar progres, but i mean, that's wotlk.
Defining a class power in T7 on a barrely 1.30min TTK, even lower on not buffed encounters, is kinda weird. With longer encounter in ulduars, things will be more balanced, around a less bursty class meta. Wotlk

**Comment by Manpower99** [comment]
Not really, it should replicate a similar amount of casts as T8 Classic Gargoyle without making it dynamic haste in my opinion.
this way we are in a position of both enjoying it but also not ruining the class by changing its playstyle. either way i've said my piece we can only await Radeghost's response now.

**Comment by Eradicatepls** [comment]
Yes or we can set it to 0 and make all us DK happy again, woohoo ! :) But yeah we wait Ragehost.

**Comment by WhitemaneIsa** [comment]
@RadeghostWM

**Comment by Karsenity** [comment]
Just wanted to add some figures that might help with decision-making on this. You can plug these functions into desmos to see a graph of how additional haste is going to affect the number of casts performed. The first figure (blue) represents the 0.2) delay while the second figure (black in the image) represent normal scaling.

One thing of note is that x*100  = the amount of additional haste players gain from gear, with things like thundering skyflare and whetstone already accounted for. I had to multiply x by  100 to make sure the graph was usable, so if you see x=1, that represents a player gaining another 1,000 haste from gear.

https://www.desmos.com/calculator

\frac{25}{\left(\frac{2}{\left(\left(1\ +\frac{x\cdot100}{2521}\right)\ +\ 0.7989\right)\ \cdot2.7939}\ +\ 0.2\right)}

\frac{25}{\left(\frac{2}{\left(\left(1\ +\frac{x\ \cdot100}{2521}\right)\ +\ 0.7989\right)\ \cdot2.7939}\right)}

![image](https://github.com/user-attachments/assets/e0512835-291a-4b50-8ee1-9a1573396a71)


The important info this tells us is that currently gargoyles are around 43 casts, with one more cast taking around 160 haste rating. This means the scaling is pretty low, and its going to continue requiring more and more haste for 1 more cast as time goes on.

If you were to remove the delay, the number of casts currently goes up to around 68-72 (depending on how much haste the DK has stacked already). To get 1 more cast is going to take around 70 haste, with the scaling being pretty much linear. 

So the result is current gargoyle would be doing about 60% more casts, and its value from haste would double. This also greatly increases the value of AP on the gargoyle, as adding 1 damage to each shot holds way more value if there's more shots. As for how this scales into Ulduar and later phases, its hard to say without doing a lot more math to see what haste/AP values look like during each raid tier. 


Honestly, if the delay was removed and it ended up being OP again, the better solution would probably just be to either increase the base cast time. My reasoning for this is that if you increase base cast time, the default gargoyle ends up being weaker than OG but stronger than current, and it requires more haste per additional cast meaning you limit its power in phase 1 and require more haste in future phases to gain additional casts. However the benefit of this is that you fix the problem with the current solution of haste having diminishing returns. 

Here's an example of what happens if you increase the cast-time to 2.5 seconds instead of the 2 it currently is. Blue is current implementation, black is the 2.5 second cast.
![image](https://github.com/user-attachments/assets/b76223a7-212d-4699-8b85-c35d544e5763)
As you can see, you end up with slightly more casts than currently but still about 20 less than retail implementation had. This alone reduces how overpowered it becomes since you don't have to deal with how insane strength scaling becomes with 75+ casts off-the-bat, but still allows players to benefit from haste scaling going forward.

**Comment by Manpower99** [comment]
Hard disagree with increasing base cast time, or the ap scaling. a delay imo is better than anything, just depends on how much.

**Comment by Karsenity** [comment]
Can I ask why you disagree? Considering you started by saying a slight buff to gargoyle would be nice going into ulduar, I'm confused why you'd prefer a method that both does less damage and is also the reason it has scaling issues to begin with.

**Comment by Manpower99** [comment]
changing base cast time and/or ap scaling will be affecting pvp. changing delay only affects pve since it just makes their haste scaling reduced.

**Comment by Karsenity** [comment]
Both affect it in pvp. Without literally any haste scaling, 2 seconds is about 12 casts, delay is 11, and 2.5 seconds is 10 casts. No matter which solution a raw gargoyle always loses damage in pvp. 

However there will never be a scenario like this in pvp, as unholy dk's always run frost subspec for lichborne meaning they would have the 20% haste + engineering gloves. Not to mention you can still get multiple other haste snapshot buffs in pvp, basically everything except bloodlust and a speed pot in an arena. At that point because the gargoyle scales better with haste without a delay, the 2.5 second cast time is probably going to reach roughly equivalent casts as the current implementation, so there really is not much of a change in pvp vs now.

**Comment by Manpower99** [comment]
No Lust no Potion, no 20% melee haste aura from FDK/Enha, no 3% aura either. no matter what a UHDK in pvp does not reach the same amount of casts ever. their gear is optimized towards PVP also. a 2.5s cast time is a pure nerf to uhdk in pvp. it is already easily countered and this will just make it poopoo.

**Comment by Karsenity** [comment]
Also I think its important to recognize that the main reason for this suggestion is realistic expectations for the devs. We can't expect them to change gargoyle every tier, if they alter the delay for ulduar so it scales better then they'll just have the same problem of it being overtuned in naxx when next season starts.

Literally the possible outcomes here are they don't change it and it continues to struggle with scaling in later tiers, they reduce/remove the delay and we repeat this whole thing in a year, or they make a change like this that addresses both issues simultaneously as best as they can.

**Comment by Karsenity** [comment]
> No Lust no Potion, no 20% melee haste aura from FDK/Enha, no 3% aura either. no matter what a UHDK in pvp does not reach the same amount of casts ever. their gear is optimized towards PVP also. a 2.5s cast time is a pure nerf to uhdk in pvp. it is already easily countered and this will just make it poopoo.

Can you do the math to provide evidence that this is the case then?

**Comment by Manpower99** [comment]
?? this isn't pve where gargoyle simply sits and casts. it is a caster guardian which you can LoS, silence and interrupt CC through multiple ways. a 0.5 extra cast time will just make this even worse than it already is. and yes i'm sure theres alot of Unholy dks who team up with enhas and frost dks lol.

**Comment by Karsenity** [comment]
So doing some quick napkin math:

20% haste from frost
20% from windfury
~30% haste from engi gloves + meta
30% from troll racial

= 1.2 * 1.2 * 1.3 * 1.3 = 2.43

2/2.43 = 0.82 cast time + 0.2 delay = 1.02 time between casts

2.5/2.43 = 1.03 cast time

They're literally the same number of casts. The only difference is 0.2 more seconds to react and pummel the latter.

**Comment by Eradicatepls** [comment]
Karsenity, ty for your time and your clear explanations about your way of seeing things. 
Really liked the details and the maths resources. I'm mostly agreeing with you. I think it's a good idea too, let's see what are thinking the development team here.

I think we could set cast time to 2.25 thought, i think 2.50 is too less of a change, but this can be discussed later on, and that mostly because i'm acknowledging the late ICC game content, where DK starts to really be hard outscaled and disappear from the Busted CLASS roaster.

Remember also that we're playing on a Buffed realm, where we'll obviously never come close to Wolk Classic TTK of top parsing guild. 
DK was also god of classic because of DKs stacking and parsing set up with 2/3 healers.  
OFC  with 12 DKs and 30s encounters you can abuse the game if not tweaked correctly.

Impossible to mimic this on Whitemane frostmourne, we're not in the warcraft log meta, with less players also, no real reasons to reunite this kind of raid set up, plus harder to do with bosses having more HP Pool, exept maybe for 1 or 2 guilds.

Uh is the best dps of WOTLK within the right conditions, it can give you acces to certain dps requirement due to it's bursty nature, (like Anub 1 phase for exemple.) but for the most guilds, with normal TTK, it's not gonna unlock DK stacking day 1 in ulduar for your random standard withemane guild. Content will remain the same. Hard and long.

ATM the best kill on patchwerk is 1h50 avrg here, on classic was 1m10. The difference in term of TTK is already balancing more the game, and it will be even more the case with the beginning of Ulduar.

**Comment by Sylanha** [comment]
![image](https://github.com/user-attachments/assets/7c42a27b-9a60-46c9-beef-2947f437ed6d)
![image](https://github.com/user-attachments/assets/2a1af0f3-6947-4042-852e-e606d2a2303b)


I dont really get the whole idea of "balancing gargoyle" right now. As you can see on the picture above there are more warlocks. In fact 50% of the current Top10 Patchwerk/Sapphiron kills are warlocks. The dps difference is so low + we are basically at the end of Tier 7 right now. Everyone in Top10 pretty much has full BIS and is in a semi-hardcore or parsing guild. On Patchwerk #1 to #10 is within 200dps ish, which lets me wonder what's up with the whole obsession with nerfing one class and spec in particular? Compared to classic you only saw Unholy DKs since as mentioned (by Manpower i think) the kill timers were even shorter on classic and they were actually stacking Unholy DKs unlike on Whitemane stacking Warlocks. Meanwhile on Whitemane, even in current T7 which to every semi-hardcore guild is considered a "joke"-Tier, you still don't see Unholy being stacked, because of an opressive gargoyle, no? Instead you are still seeing Warlock and that's on top of the parsing culture since Naxx is easy. 

In the Patchwerk kill that I listed above are the current Top10 PLAYERS with short kill timer. That doesn't represent the average Whitemane Player in the slightest, meaning that the whole idea of nerfing Unholy is really questionable. With the upcoming release of Ulduar for example, the difficulty level is going to inrease by a lot, especially for the average Whitemane Player that doesn't play in a guild or plays in a dad-friendly casual guild. Even in semi-hardcore Guilds you won't get away with stacking Unholy, because you simply can't afford it. It also just won't make much sense, since right now you don't even notice it. Why would you see it on much harder content + longer kill timers?

Don't forget that Unholy is supposed to be a bursty and S-Tier class in earlier Tiers. It's going to lose A LOT of the haste value which is going to lea to more problems trying to balance around the casting bar or haste scaling, since it's not linear compared to other classes.

If you change how gargoyle is working right now, it would mean you would need to adjust it again in future Tiers, especially in T9-T10 when Gargoyle has lost basically 1700 haste from all the snapshotting you currently have in T7.

My suggestion would be to either decrease the 200MS delay as suggested above to maybe 100 or even go full back and revert the random delay since it's going to affect later Tiers more than the current tier right now.
A class being "opressive or just OP" would make me think that giving a Patchwerk scenario + short kill timers would make it stand out multiple 2-3k dps in front of any other class. MEANWHILE right now every warlock in T10 which makes 50% of the total Top10 is doing 10k DPS on a 1:40min - 2min kill timer (which is extremely short especially considering it's single target aka no multi dotting + no longer execution drain uptime, compared to a S-Tier Burst class that is supposed to outDPS Warlocks with those conditions (look at WarcraftLogs 2023 and compare the Infernals of Warlock, they did way more compared to Whitemane) 

There is no need to artificially gutter a class because of spite or different opinions from the uproar of players that aren't justified (again look at Top10 Patchwerk fights and then think about the average Whitemane Player).

TL;DR: Revert the 200MS delay change to 0 or 100 and just wait. The 2.5sec base delay sounds terrible and would potentially just cause more issues especially when the answer seems pretty clear looking at current Top10 DPS being very evenly.

**Comment by crabcakeowo** [comment]
just to be clear, dks are fine where they are right now.
![image](https://github.com/user-attachments/assets/532cf38b-47a0-4e96-864e-9332c5b16ca6)

this is our run at 1:10 with our DK having chicken equipped instead of an actual DPS trinket. Balanced? yeah i'd like to think it is, anyone who says otherwise clearly are biased towards buffing dks to the moon.

**Comment by crabcakeowo** [comment]
And also about balancing gargoyle, it should have been done with dynamic haste with 0 cast delay like what blizzard did, there was a classic warcraftlog pulled up in togc where gary did 49 casts in one summon compared to our 44-45ish casts now.

However that takes the fun and skill aspect out of playing unh dk so whitemane decided not to go that route, so having it at its current state is where they felt is balanced.

**Comment by Manpower99** [comment]
> ![image](https://private-user-images.githubusercontent.com/172520001/357560603-7c42a27b-9a60-46c9-beef-2947f437ed6d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5NDAwNjgsIm5iZiI6MTcyMzkzOTc2OCwicGF0aCI6Ii8xNzI1MjAwMDEvMzU3NTYwNjAzLTdjNDJhMjdiLTlhNjAtNDZjOS1iZWVmLTI5NDdmNDM3ZWQ2ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxOFQwMDA5MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMTZjMTFiZmVmNGMyYTY3NjlkN2I0ODZlMWQzZjczODRhZGEwYjZiNzg0ZDhmMGY4NTgzNGU0MWViODZlNDRmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.O--DcMuEso8mZ1NzOcX7sJJB1Xv0uIH1NUttWUHT65k) ![image](https://private-user-images.githubusercontent.com/172520001/357617963-2a1af0f3-6947-4042-852e-e606d2a2303b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM5NDAwNjgsIm5iZiI6MTcyMzkzOTc2OCwicGF0aCI6Ii8xNzI1MjAwMDEvMzU3NjE3OTYzLTJhMWFmMGYzLTY5NDctNDA0Mi04NTJlLWU2MDZkMmEyMzAzYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODE4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgxOFQwMDA5MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMTU1MjVkN2FhM2FiNTJhNzNiNzNlOTQ5ZWIyZGMzNmQ2OGU1YjE0M2VmZTFhZWFiNWY1NzMxYzQ0YWE5NDQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.quCDwhoUQ_77tlqY-CWjvydqfwBz7TgnoWH84TX0pa0)
> 
> I dont really get the whole idea of "balancing gargoyle" right now. As you can see on the picture above there are more warlocks. In fact 50% of the current Top10 Patchwerk/Sapphiron kills are warlocks. The dps difference is so low + we are basically at the end of Tier 7 right now. Everyone in Top10 pretty much has full BIS and is in a semi-hardcore or parsing guild. On Patchwerk #1 to #10 is within 200dps ish, which lets me wonder what's up with the whole obsession with nerfing one class and spec in particular? Compared to classic you only saw Unholy DKs since as mentioned (by Manpower i think) the kill timers were even shorter on classic and they were actually stacking Unholy DKs unlike on Whitemane stacking Warlocks. Meanwhile on Whitemane, even in current T7 which to every semi-hardcore guild is considered a "joke"-Tier, you still don't see Unholy being stacked, because of an opressive gargoyle, no? Instead you are still seeing Warlock and that's on top of the parsing culture since Naxx is easy.
> 
> In the Patchwerk kill that I listed above are the current Top10 PLAYERS with short kill timer. That doesn't represent the average Whitemane Player in the slightest, meaning that the whole idea of nerfing Unholy is really questionable. With the upcoming release of Ulduar for example, the difficulty level is going to inrease by a lot, especially for the average Whitemane Player that doesn't play in a guild or plays in a dad-friendly casual guild. Even in semi-hardcore Guilds you won't get away with stacking Unholy, because you simply can't afford it. It also just won't make much sense, since right now you don't even notice it. Why would you see it on much harder content + longer kill timers?
> 
> Don't forget that Unholy is supposed to be a bursty and S-Tier class in earlier Tiers. It's going to lose A LOT of the haste value which is going to lea to more problems trying to balance around the casting bar or haste scaling, since it's not linear compared to other classes.
> 
> If you change how gargoyle is working right now, it would mean you would need to adjust it again in future Tiers, especially in T9-T10 when Gargoyle has lost basically 1700 haste from all the snapshotting you currently have in T7.
> 
> My suggestion would be to either decrease the 200MS delay as suggested above to maybe 100 or even go full back and revert the random delay since it's going to affect later Tiers more than the current tier right now. A class being "opressive or just OP" would make me think that giving a Patchwerk scenario + short kill timers would make it stand out multiple 2-3k dps in front of any other class. MEANWHILE right now every warlock in T10 which makes 50% of the total Top10 is doing 10k DPS on a 1:40min - 2min kill timer (which is extremely short especially considering it's single target aka no multi dotting + no longer execution drain uptime, compared to a S-Tier Burst class that is supposed to outDPS Warlocks with those conditions (look at WarcraftLogs 2023 and compare the Infernals of Warlock, they did way more compared to Whitemane)
> 
> There is no need to artificially gutter a class because of spite or different opinions from the uproar of players that aren't justified (again look at Top10 Patchwerk fights and then think about the average Whitemane Player).
> 
> TL;DR: Revert the 200MS delay change to 0 or 100 and just wait. The 2.5sec base delay sounds terrible and would potentially just cause more issues especially when the answer seems pretty clear looking at current Top10 DPS being very evenly.

Fully agreed.

**Comment by crabcakeowo** [comment]
https://classic.warcraftlogs.com/reports/bLvhrPdTNXZz84H1#fight=16&type=damage-done&source=9

There's this myth that everyone is saying that unholy dks lose all their haste, this log proves that even with a dynamic hasted gary it was still casting 43 shots.

I don't see why any change is needed from its current state?

Also, if you're going to make an argument that it wasn't like this during Naxxramas, if you're going to get a 350k damage gargoyle during naxx and a 220k gargoyle during TOC then your class shouldn't be scaling in the first place.

**Comment by UltraNix** [comment]
Okay guys - we'lll change to 100ms and see what happens :)

**Comment by UltraNix** [comment]
Change applied - waiting for some feedback.

**Comment by Manpower99** [comment]
> Change applied - waiting for some feedback.

Will do once we get more Ulduar logs. currently its very limited.

**Comment by danniellos94** [comment]
I'm playing pvp only here so I can tell from pvp perspective it's the same, could literally be 0 ms and won't be a problem.

**Comment by Manpower99** [comment]
> Change applied - waiting for some feedback.

Yeah its noticably a pretty nice change, didnt affect PVP like the other suggestion would've done. instead pushed the spec to be more more enjoyable in regards to trinkets/haste scaling. would recommend it stays this way with no more changes needed.
